src/widgetastic_patternfly/__init__.py

```diff
@@ -1804,3 +1805,140 @@ def items(self):
+                el.click()
+                return True
+
+    def _select_year(self, year):
+        for yr in range(1, 10):
+            start_yr, end_yr = [int(item) for item in self.date_switch_value.split('-')]
+            if start_yr <= int(year) <= end_yr:
+                for el in self.browser.elements(self.YEAR):
+                    if int(el.text) == int(year):
+                        el.click()
+                        return True
+            elif int(year) < start_yr:
+                self.browser.click(self.PREV)
+            elif int(year) > end_yr:
+                self.browser.click(self.NEXT)
+        else:
+            raise ValueError("Year not valid")
+
+    def read(self):
+        return self.value
+
+    def select(self, date):
+        """select date
+
+        Args:
+            date (dict): date in `dict` format with keys `day`, `month`, `year`
+        Returns:
+            :py:class:`bool`
+        """
+        self.parent_browser.click(self.TEXT_BOX)
+        date = datetime(date['year'], date['month'], date['day'])
+        month = date.strftime("%b")
+        day = date.strftime("%d")
+        year = date.strftime("%Y")
+        sw_value = self.date_switch_value.split(' ')
+
+        if (date.strftime("%B") not in sw_value) or (year not in sw_value):
+            self.browser.click(self.DATE_PICKER_SWITH)
+            if year not in self.date_switch_value:
+                self.browser.click(self.DATE_PICKER_SWITH)
+                self._select_year(year)
+            self._select_month(month)
+        self._select_date(day)
+        return True
+
+    def fill(self, value):
+        """Fill date to date box
+
+        Args:
+           value (str/ dict): date in `dict` format having keys `day`, `month`, `year` or
+           date in `str` format `dd/mm/yyyy`
+        Returns:
+            :py:class:`bool`
+        """
+        date_mapp = self.date_format
+        for item in self.date_format.split('/'):
+            date_mapp = date_mapp.replace(item, self.DATE_MAPPING[item])
+
+        if isinstance(value, str):
+            date_obj = datetime.strptime(value, "%m/%d/%Y")
+        elif isinstance(value, dict):
+            date_obj = datetime(value['year'], value['month'], value['day'])
+        else:
+            ValueError("Value should be in string (dd/mm/yyyy) or \
+            dictionary having keys day, month, year")
+
+        date = datetime.strftime(date_obj, date_mapp)
+
+        if date == self.value:
+            return False
+
+        if not self.is_readonly:
+            self.parent_browser.clear(self.TEXT_BOX)
+            self.parent_browser.send_keys(date, self.TEXT_BOX)
+            return True
+        else:
+            return self.select(
+                {'day': date_obj.day, 'month': date_obj.month, 'year': date_obj.year})
+
+    @property
+    def is_displayed(self):
+        return self.parent_browser.is_displayed(self.TEXT_BOX)
+
+    @property
+    def is_readonly(self):
```

Comment:
REQUIRED: all public methods and properties should have docstrings with description